app/components/show/catkey_component.rb

```diff
@@ -2,16 +2,16 @@
 
 module Show
   class CatkeyComponent < ApplicationComponent
-    def initialize(document:, state_service:)
-      @document = document
+    def initialize(change_set:, state_service:)
+      @change_set = change_set
       @state_service = state_service
     end
 
     def catkey
-      catkey_id || 'None assigned'
+      @change_set.catkey.presence || 'None assigned'
     end
 
     delegate :allows_modification?, to: :@state_service
-    delegate :id, :catkey_id, to: :@document
+    delegate :id, to: :@change_set
```

Comment:
Not a big deal, but I wonder when (and when not) we want to delegate, specifically wondering about `:catkey`.

Reply:
In this case we don't delegate, because we display a default value when the model has no catkeys.